hibernate-core/src/main/java/org/hibernate/boot/internal/MetadataBuilderImpl.java

```
@@ -118,10 +118,12 @@ private static StandardServiceRegistry getStandardServiceRegistry(ServiceRegistr
 			return (StandardServiceRegistry) serviceRegistry;
 		}
 		else if ( serviceRegistry instanceof BootstrapServiceRegistry ) {
-			log.debug(
-					"ServiceRegistry passed to MetadataBuilder was a BootstrapServiceRegistry; this likely won't end well " +
-							"if attempt is made to build SessionFactory"
-			);
+			if (log.isDebugEnabled()) {
+				log.debug(
+						"ServiceRegistry passed to MetadataBuilder was a BootstrapServiceRegistry; this likely won't end well " +
+								"if attempt is made to build SessionFactory"
+				);
+			}
```

Unless this log message is wrong and incorrect, an exception should be thrown.

---

there is an attempt to resolve this 'gracefully' here though, which makes me think a warn level is more adequate here.

---

"Gracefully" how.

The log message says it will blow up later.

And blowing up earlier is always better than blowing up later.

---

Failing "gracefully" usually means failing fast as soon as possible with a useful error message

---

There's clearly the intention to try and go ahead in spite of a potential situation, otherwise there's wouldn't be the `new StandardServiceRegistryBuilder( (BootstrapServiceRegistry) serviceRegistry ).build()` below
I didn't check, but maybe an exception will be thrown further down the line?

---

> maybe an exception will be thrown further down the line?

Well, that's what the log message claims. Perhaps the log message is wrong.